huaweicloud/services/dds/resource_huaweicloud_dds_database_role.go

```diff
@@ -0,0 +1,326 @@
+		DbName: d.Get("db_name").(string),
+	}
+	resp, err := roles.List(client, instanceId, opts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, fmt.Sprintf("error getting database role (%s) from DDS instance (%s)",
+			name, instanceId))
+	}
+	if len(resp) < 1 {
+		return common.CheckDeletedDiag(d, err, fmt.Sprintf("unable to find database role (%s) from DDS instance (%s)",
+			name, instanceId))
+	}
+	role := resp[0]
+	log.Printf("[DEBUG] The role response is: %#v", role)
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", role.Name),
+		d.Set("db_name", role.DbName),
+		d.Set("roles", flattenDatabaseRoles(role.Roles)),
+		d.Set("privileges", flattenDatabasePrivileges(role.Privileges)),
+		d.Set("inherited_privileges", flattenDatabasePrivileges(role.InheritedPrivileges)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func databaseRoleRefreshFunc(client *golangsdk.ServiceClient, instanceId, roleName,
+	dbName string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		opts := roles.ListOpts{
+			Name:   roleName,
+			DbName: dbName,
+		}
+		resp, err := roles.List(client, instanceId, opts)
+		if err != nil {
+			return nil, "ERROR", fmt.Errorf("error getting database role (%s) from DDS instance (%s)", roleName,
+				instanceId)
+		}
+		if len(resp) < 1 {
+			return resp, "DELETED", nil
+		}
+		return resp, "ACTIVE", nil
+	}
+}
+
+func resourceDatabaseRoleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.DdsV3Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DDS v3 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	config.MutexKV.Lock(instanceId)
+	defer config.MutexKV.Unlock(instanceId)
+	name := d.Get("name").(string)
+	dbName := d.Get("db_name").(string)
+	opts := roles.DeleteOpts{
+		Name:   name,
+		DbName: dbName,
+	}
+	err = roles.Delete(client, instanceId, opts)
+	if err != nil {
+		return diag.Errorf("error deleting database role (%s) from DDS instance (%s): %v", name, instanceId, err)
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"ACTIVE"},
+		Target:       []string{"DELETED"},
+		Refresh:      databaseRoleRefreshFunc(client, instanceId, name, dbName),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        2 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for database role deletion to complete: %s", err)
+	}
+
+	return nil
+}
+
+func resourceDatabaseRoleImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID, must be <instance_id>/<name>")
+	}
+
+	d.Set("instance_id", parts[0])
+	d.Set("name", parts[1])
```

Comment:
how about move this part to Read func, then we don't need this func and testAccDatabaseRoleImportStateIdFunc

Reply:
I think it is enough to keep the current method, but the ID currently has a problem that it is not unique and needs to be modified.